tests/test_streaming.py

```diff
@@ -59,6 +59,65 @@ def test_dataloader_epoch_size_no_streams(local_remote_dir: Tuple[str,
             assert samples_seen == epoch_size
 
 
+@pytest.mark.parametrize('batch_size', [4])
```

Comment:
Can you add a `batch_size` which is non-divisible of `num_samples/stream` and validate if the last samples are being dropped or not or is this already being handled by other params such as `num_workers` and `num_canonical_nodes` ?

Reply:
done!